## Welcome to the FUNKI application
FUNKI is a multi-omic functional integration and analysis platform. It provides a standardise pipeline to process and perform functional analysis on transcriptomic, proteomic, phosphoproteomic and metabolomic datasets. The analysis can be performed both on single type of omic data and on multi-omic dataset by integrating them in supervised and unsupervised manners.

### Implemented Approaches
The following approaches are implemented in the FUNKI workflow

#### DoRothEA
DoRothEA is a gene regulatory network linking Transcription Factors (TFs) with their downstream targets. The unity of TF and its targets is referred as regulon. Such gene regulatory networks allow the computation of TF-activities from gene expression data by enrichment analysis using the regulons as underlying gene-sets. The network has been build integrating TF-target interactions from four different strategies: (i) manually curated interaction repositories, (ii) interactions derived from ChIP-seq binding data, (iii) in silico predictions of TF binding on gene promotors, and (iv) reverse-engineered regulons from large gene expression datasets. Based on the number of supporting evidences each interaction is accompanied with a confidence level ranging from high-confident (A) to low-confident (E). More details can be found [here](https://www.biorxiv.org/content/early/2018/06/18/337915).

#### PROGENy
[Coming soon...](https://saezlab.github.io/progeny/)

#### KinAct
[Coming soon...](http://saezlab.github.io/kinact/)

